### ISSUE TITLE
Kernel build: arch/x86/tools/insn_decoder_test: error: malformed line

### ISSUE THREAD

**Issue by dhodovsk** [ClangBuiltLinux/linux#2060]
While testing the mainline kernel, we get an error during the x86_64 build:
```
00:29:20 arch/x86/tools/insn_decoder_test: error: malformed line 3733181:
00:29:20 2_>:ffffffff81c92050
00:29:20 make[4]: *** [arch/x86/tools/Makefile:26: posttest] Error 3
00:29:20 make[3]: *** [arch/x86/Makefile:310: bzImage] Error 2
00:29:20 make[3]: *** Waiting for unfinished jobs....
```

For us, it started happening two weeks ago, since the [Merge tag 'rust-6.13' of https://github.com/Rust-for-Linux/linux](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=798bb342e0416d846cf67f4725a3428f39bfb96b) commit in kernel mainline.

The issue was already discussed in [linux-kernel Mailing List](https://lore.kernel.org/lkml/Y9ES4UKl%2F+DtvAVS@gmail.com/T/), but it seems the suggested patches didn't come through.

Links:
- our pipeline: https://gitlab.com/redhat/red-hat-ci-tools/kernel/cki-internal-pipelines/cki-trusted-contributors/-/jobs/8607023441
- the full build log: https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/build.log
- kernel config: https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/kernel-upstream-mainline-redhat_1583189702-x86_64-kernel.config

I'm sorry if this is not the right place to report the issue. I'd be glad for any pointers.

**Comment by nickdesaulniers** [comment]
Is that a rust mangled symbol? Maybe the rust for Linux project should
receive this bug report?

**Comment by nathanchance** [comment]
Yeah, this seems to be related to Rust, not LLVM/Clang:

https://bugzilla.redhat.com/show_bug.cgi?id=2329496
https://lore.kernel.org/20241129154031.GA7195@redhat.com/

That ping is relatively recent but I think it would probably be good to open an issue on [the Rust-for-Linux issue tracker](https://github.com/Rust-for-Linux/linux/issues) with clear instructions on how to reproduce, such as:

```
$ git show -s --format='%h ("%s")'
f92f4749861b ("Merge tag 'clk-fixes-for-linus' of git://git.kernel.org/pub/scm/linux/kernel/git/clk/linux")

$ rustc --version
rustc 1.83.0 (90b35a623 2024-11-26)

$ clang --version | head -1
ClangBuiltLinux clang version 19.1.5 (https://github.com/llvm/llvm-project.git ab4b5a2db582958af1ee308a790cfdb42bd24720)

$ curl -LSso .config https://s3.amazonaws.com/arr-cki-prod-trusted-artifacts/trusted-artifacts/1583189702/build_x86_64/8607023441/artifacts/kernel-upstream-mainline-redhat_1583189702-x86_64-kernel.config

$ make -skj"$(nproc)" ARCH=x86_64 LLVM=1 olddefconfig bzImage
...
arch/x86/tools/insn_decoder_test: error: malformed line 3735255:
2_>:ffffffff81c94190
make[4]: *** [arch/x86/tools/Makefile:26: posttest] Error 3
...
```

**Comment by nathanchance** [comment]
There was another patch sent to fix this a few weeks ago as well:

https://lore.kernel.org/20241120-insn-decoder-test-len-fix-v1-1-44b075deda05@gmail.com/

**Comment by dhodovsk** [comment]
The patch proposed in https://lore.kernel.org/20241120-insn-decoder-test-len-fix-v1-1-44b075deda05@gmail.com/ is similar to the one from last January, which seems to get lost in the discussion of whether the length would be still okay.

As you advised, I will open an issue in https://github.com/Rust-for-Linux/linux/issues.

Thank you both for your quick answers and the reproducer! I hope I will learn what to report where ASAP :)

**Comment by nickdesaulniers** [comment]
Happy to help. Better to report than not.
